feat(tracing): add docs for AI SDK 7 (#4623)
langchain-ai/docs: src/langsmith/trace-with-vercel-ai-sdk.mdx

---
title: Trace Vercel AI SDK applications (JS/TS only)
sidebarTitle: Vercel AI SDK
description: Trace Vercel AI SDK applications with LangSmith for AI SDK v5, v6, and v7.
---

> You can use LangSmith to trace runs from the Vercel AI SDK. This guide shows how to set up tracing for AI SDK v5, v6, and v7.

Reviewer: ```suggestion
You can use LangSmith to trace runs from the Vercel AI SDK. This guide shows you how to set up tracing for AI SDK v5, v6, and v7.
```

## Installation

> Install the Vercel AI SDK, a model provider package, and LangSmith. This guide uses Vercel's OpenAI integration for the code snippets below, but you can use any other Vercel AI SDK provider.

Reviewer: ```suggestion
Install the Vercel AI SDK, a model provider package, and LangSmith. This guide uses Vercel's OpenAI integration for the following code snippets, but you can use any other Vercel AI SDK provider.
```

<Tabs>
    <Tab title="AI SDK v7">
        <Note>

>         `LangSmithTelemetry` requires AI SDK v7. If you are using an older version of the AI SDK or `langsmith`, see the OpenTelemetry (OTEL) based approach [on this page](/langsmith/legacy-trace-with-vercel-ai-sdk).

Reviewer: ```suggestion
        `LangSmithTelemetry` requires AI SDK v7. If you are using an older version of the AI SDK or `langsmith`, refer to the [OpenTelemetry (OTEL) based approach](/langsmith/legacy-trace-with-vercel-ai-sdk).
```

        </Note>

        <CodeGroup>

        ```bash npm
        npm install ai @ai-sdk/openai zod langsmith
        ```

        ```bash yarn
        yarn add ai @ai-sdk/openai zod langsmith
        ```

        ```bash pnpm
        pnpm add ai @ai-sdk/openai zod langsmith
        ```

        </CodeGroup>
    </Tab>
    <Tab title="AI SDK v5 and v6">
        <Note>

>         `wrapAISDK` supports AI SDK v6 and requires `langsmith>=0.7.2`. If you are using an older version of the AI SDK or `langsmith`, see the OpenTelemetry (OTEL) based approach [on this page](/langsmith/legacy-trace-with-vercel-ai-sdk).

Reviewer: ```suggestion
        `wrapAISDK` supports AI SDK v6 and requires `langsmith>=0.7.2`. If you are using an older version of the AI SDK or `langsmith`, refer to the [OpenTelemetry (OTEL) based approach](/langsmith/legacy-trace-with-vercel-ai-sdk).
```
